core/src/main/java/org/aerogear/mobile/core/MobileCore.java

```diff
@@ -145,7 +145,7 @@ public void destroy() {
             ServiceModule serviceModule = serviceClass.newInstance();
 
             if (serviceConfiguration == null) {
-                serviceConfiguration = getServiceConfiguration(serviceModule.type());
+                serviceConfiguration = getServiceConfiguration(serviceModule.type(), serviceModule.requiresConfiguration());
```

Comment:
I think it would be better to leave the 
`getServiceConfiguration(serviceModule.type())` as it is, without the `requiresConfiguration` option and simply making it returns null if no configuration exists.
Then here we could do:
`if (config == null && serviceModule.requiresConfiguration()) {
    throw new ConfigurationNotFoundException(...)`
WDYT?

Reply:
I'm happy to allow the service configuration to be null and have each service handle what happens in that scenario. @danielpassos @wei-lee Would you be fine with that?

Reply:
@aidenkeating that's not exactly what I said. Only the service knows if they need or not a configuration, however, if they need it, it is managed centrally inside the `getInstance` method of the `MobileCore`. From the outside the result is the same, but IMO the code is cleaner this way (because from the POV of the `getConfiguration` method, not finding a configuration is not an exceptional case, so no exception should be raised).

Reply:
@ziccardi Apologies, misunderstanding, will make this change